File: transpiler/javatests/com/google/j2cl/transpiler/readable/jsmemberinnativetype/MyNativeType.java
package com.google.j2cl.transpiler.readable.jsmemberinnativetype;

import jsinterop.annotations.JsOverlay;
import jsinterop.annotations.JsType;

@JsType(isNative = true, namespace = "test.foo")
public class MyNativeType {
  public int publicField;
  private int privateField;
  int packageField;
  protected int protectedField;

  public native void publicMethod();

  private native void privateMethod();

  native void packageMethod();

  protected native void protectedMethod();

  @JsOverlay
  public final void useFieldsAndMethods() {
    @SuppressWarnings("unused")
    int value = publicField + privateField + packageField + protectedField;
    publicMethod();
    privateMethod();
    packageMethod();
    protectedMethod();
  }
}

File: transpiler/javatests/com/google/j2cl/transpiler/readable/jsmemberinnativetype/native_sources/MyNativeType.js
goog.module('test.foo.MyNativeType');

class MyNativeType {
  constructor() {
    this.publicField = 0;
    this.privateField = 0;
    this.packageField = 0;
    this.protectedField = 0;
  }

  /**
   * @public
   */
  publicMethod() {}

  /**
   * @public
   */
  privateMethod() {}

  /**
   * @public
   */
  packageMethod() {}

  /**
   * @public
   */
  protectedMethod() {}
}

exports = MyNativeType;
